core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/resourcePolicies/ResourceAllocator.scala

```diff
@@ -82,6 +88,43 @@ class DefaultResourceAllocator(
     (region.copy(resourceConfig = Some(resourceConfig)), 0)
   }
 
+  private def createOutputPortStorageObjects(region: Region): Unit = {
+    // Create storage objects
+    region.storageURIs.foreach {
+      case (outputPortId, storageUriToAdd) =>
+        val (wid, eid, _, _, _, _) = decodeURI(storageUriToAdd)
+        val existingStorageUri =
+          WorkflowExecutionsResource.getResultUriByExecutionAndPort(
+            wid = wid,
+            eid = eid,
+            opId = outputPortId.opId.logicalOpId,
+            layerName = Some(outputPortId.opId.layerName),
+            portId = outputPortId.portId
+          )
+        if (
+          (!AmberConfig.isUserSystemEnabled && !ExecutionResourcesMapping
+            .getResourceURIs(eid)
+            .contains(
+              existingStorageUri
+            )) || (AmberConfig.isUserSystemEnabled && existingStorageUri.isEmpty)
+        ) {
```

Comment:
change to early termination? e.g.,
```
if (xxx) {
return
}
// do something here
```

Reply:
If we use return here, it will directly exit the function.